transport/src/main/java/io/netty/channel/FixedRecvByteBufAllocator.java

```diff
@@ -38,6 +41,22 @@ public int guess() {
         }
     }
 
+    /**
+     * Creates a new {@link FixedRecvByteBufAllocator} that will continue reading even when the read
+     * {@link io.netty.channel.socket.DatagramPacket} was smaller then the given {@code bufferSize}.
+     *
+     * @param bufferSize the buffer size that is used for {@link io.netty.channel.socket.DatagramPacket}s.
+     * @return the allocator.
+     */
+    public static FixedRecvByteBufAllocator newDatagramRecvByteBufAllocator(int bufferSize) {
+        FixedRecvByteBufAllocator recvAlloc = new FixedRecvByteBufAllocator(bufferSize);
+        // Let's use false here as the number of bytes may be less then what we tried to read when using datagrams.
+        // This is expected and we should tolerate this as otherwise we may need to wait for the selector to wakeup
+        // again for a read.
+        recvAlloc.respectMaybeMoreData(false);
```

Comment:
I wonder if we can make this the default for Netty 5.

Reply:
I think this only makes sense for Datagram to make it default.